### DISCUSSION TITLE
Make JWT authorization more consistent

### DISCUSSION THREAD

**Discussion by hleb-albau** [go-kit/kit#1144] [Q&A]
H!

I am using go-kit JWT package to provide auth for users in my services. For me, it's look very complicated one, especially for beginners. What caused me to stuck is token generation and validation leaving in different scopes. Maybe we can create separate abstraction for token generation/validation and use it to create `NewParser` **middleware**. 

- [ ] Create abstraction for token generation-validation
- [ ] Create example with jwt
- [ ] Update docs

**Comment by peterbourgon** [comment]
I don't know enough about JWT to make meaningful comments to this proposal, but are you sure such an abstraction is safe?

**Comment by hleb-albau** [comment]
Yeah, it is not related to jwt generation itself, but rather the abstractions, that go-kit creates.

**Comment by peterbourgon** [comment]
I would happily review a PR. Push step one before doing two and three so if there is a design issue you don't waste energy on code that would need to change.
